inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.cpp

```diff
@@ -0,0 +1,538 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "mkldnn_strided_slice_node.h"
+
+#include <mkldnn_types.h>
+#include <mkldnn_extension_utils.h>
+
+#include "ie_parallel.hpp"
+#include <legacy/ie_layers.h>
+#include "common/dnnl_thread.hpp"
+#include "common/cpu_memcpy.h"
+#include "utils/general_utils.h"
+#include "common/dnnl_thread.hpp"
+
+#include <string>
+#include <algorithm>
+
+#define THROW_ERROR THROW_IE_EXCEPTION << "StridedSlice layer with name '" << getName() << "' "
+
+using namespace mkldnn;
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+static inline size_t parallel_init(size_t start, size_t nDims, const SizeVector& dims, SizeVector& indexes) {
+    for (int j = nDims - 1; j >= 0; j--) {
+        indexes[j] = start % dims[j];
+        start = start / dims[j];
+    }
+    return start;
+}
+
+MKLDNNStridedSliceNode::MKLDNNStridedSliceNode(const InferenceEngine::CNNLayerPtr& layer, const mkldnn::engine& eng, MKLDNNWeightsSharing::Ptr &cache) :
+        MKLDNNNode(layer, eng, cache) {}
+
+void MKLDNNStridedSliceNode::getSupportedDescriptors() {
+    auto stridedSliceLayer = getCnnLayer();
+
+    if (stridedSliceLayer == nullptr)
+        THROW_ERROR << "cannot convert from CNN layer";
+
+    auto inData = stridedSliceLayer->insData[DATA_ID].lock();
+    auto beginData = stridedSliceLayer->insData[BEGIN_ID].lock();
+    auto endData = stridedSliceLayer->insData[END_ID].lock();
+    if (!inData || !beginData || !endData)
+        THROW_ERROR << "has nullable input data";
+
+    const SizeVector srcDims = inData->getTensorDesc().getDims();
+    const SizeVector dstDims = stridedSliceLayer->outData[0]->getTensorDesc().getDims();
+
+    if (getParentEdges().size() != 3 && getParentEdges().size() != 4)
+        THROW_ERROR << "has incorrect number of input edges";
+    if (!getChildEdges().size())
+        THROW_ERROR << "has incorrect number of output edges";
+
+    SizeVector beginDims = beginData->getTensorDesc().getDims();
+    if (beginDims.size() != 1)
+        THROW_ERROR << " should have begin vector with 1 dimension";
+
+    SizeVector endDims = endData->getTensorDesc().getDims();
+    if (endDims.size() > 1)
+        THROW_ERROR << "should have end vector with 1 dimension";
+    if (beginDims[0] != endDims[0])
```

Comment:
No because there are may be cases from [example 1](https://github.com/openvinotoolkit/openvino/blob/3b614e54c5d5de587c64e189f5477c08d58e5a34/inference-engine/src/mkldnn_plugin/nodes/mkldnn_strided_slice_node.cpp#L305) (for `dstDims.size()` too)

Reply:
Ok. Got it. In fact the behavior in case beginDims/endDims size not equal to src rank is not documented in the spec or I missed smt?

Reply:
It's not documented in the spec

Reply:
I think we need to extend the spec than. Could you please follow up this?
cc'ed @lazarevevgeny